### PR TITLE
Activity page tests in progress

### LINKED ISSUE #407: Integrate playwright tests for activity page
We should integrate following tests:

- [x] All activity tabs should be visible
- [x] Post an update
- [x] Post a Question
- [x] Post an Idea
- [x] Post a Feedback
- [x] Post a Request
- [ ] Create a Proposal
- [x] Create an event
- [x] Bookmark post
- [x] Edit post and save
- [x] Convert post into proposal
- [ ] Share/copy post link
- [x] Repost a post
- [x] Like a post
- [x] Comment on a post

### PR DESCRIPTION
**Pull request by Eren-Yeaager**
Resolves #407
1.  Added content check along with url check for Navbar tabs.

### CODE REVIEW

**Comment by Megha-Dev-19** [comment]
@Eren-Yeaager what all is left in this?
let me know if you need any help.

**Review by Megha-Dev-19 — COMMENTED** [review]
Left few suggestions.
It looks great, good job @Eren-Yeaager.

File: playwright-tests/tests/activity.spec.js [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+      transactionObj["saswat_test.testnet"].every.event,
+    )[0];
+
+    const actualEventData =
+      transactionObj["saswat_test.testnet"].every.event[dynamicEventId];
+    const expectedMetadata = {
+      name: "Test event",
+      description: "Event Description",
+      image: {},
+      backgroundImage: {},
```
[the remark this answers is not in the document]
Comment (Eren-Yeaager) [reply playwright-tests/tests/activity.spec.js:332]: Added them in a new pull request

File: playwright-tests/tests/activity.spec.js [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+    };
+
+    expect(actualEventData.metadata).toMatchObject(expectedMetadata);
+
+    const eventDataString = actualEventData[""];
+    expect(eventDataString).toContain('"title":"Test event"');
+    expect(eventDataString).toContain('"description":"Event Description"');
+    expect(eventDataString).toContain('"url":"google-meet.xyz"');
+    expect(eventDataString).toContain(
+      '"extendedProps":{"organizers":["saswat_test.testnet"],"location":"India","hashtags":[],"cover":{}}',
```
[the remark this answers is not in the document]
Comment (Eren-Yeaager) [reply playwright-tests/tests/activity.spec.js:343]: Added them in a new pull request

File: playwright-tests/tests/activity.spec.js [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+
+  test("Comment on a post", async ({ page }) => {
+    await clickAndAssertTab(page, "All", "?page=activity&tab=all", "All Feed");
+    await page.waitForTimeout(1000);
+    const commentIcon = await page.getByTitle("Comment").nth(1);
+    await page.waitForTimeout(1000);
+    await commentIcon.click();
+    const commentPost = await page.frameLocator("iframe").nth(1);
+    await commentPost.locator('textarea[name="textarea"]').fill("test comment");
+    const CommentBtn = await page.getByRole("button", { name: "Comment" });
```
[the remark this answers is not in the document]
Comment (Eren-Yeaager) [reply playwright-tests/tests/activity.spec.js:490]: Fixed it too.

File: playwright-tests/tests/activity.spec.js [hunk 4]
```diff
@@ -0,0 +1,526 @@
...
+    await clickAndAssertTab(page, "All", "?page=activity&tab=all", "All Feed");
+    await page.waitForTimeout(1000);
+    const dropdown = page.locator(".bi.bi-three-dots-vertical").nth(1);
+    await dropdown.click();
+    await page.waitForTimeout(1000);
+    const convertPost = page.getByText("Propose");
+    await convertPost.click();
+    await page.waitForTimeout(1000);
+    const DAOContractID = await page.getByText("build.sputnik-dao.near");
+    await expect(DAOContractID).toBeVisible();
```
[the remark this answers is not in the document]
Comment (Eren-Yeaager) [reply playwright-tests/tests/activity.spec.js:524]: Opening a separate ticket for this as we dont have txn for proposal in testnet

**Pull request closed without merging**